src/MergeContext.js

```diff
@@ -454,6 +454,202 @@ class BranchPosition
     }
 }
 
+// Forward iterator for fields in the 'name:value' format.
+class FieldsTokenizer
+{
+    // input string and an array of allowed attributes
+    constructor(str, allowedAttributes) {
+        this._lines = str.split('\n');
+        // supported attributes
+        this._attributes = allowedAttributes;
+    }
+
+    // Searches for a line matching the 'attrName:attrValue' pattern,
+    // where attrName is one of the allowed attributes.
+    // All parsed lines are removed from the input.
+    // Returns the matched {attrName, attrValue} pair (or null).
+    nextField() {
+        while (!this.atEnd()) {
+            const line = this._lines.shift();
+            const pos = line.search(':');
+            if (pos >= 0) {
+                const name = line.substring(0, pos);
+                if (this._attributes.includes(name))
+                    return {name: name, value: line.substring(pos+1)};
+            }
+        }
+        return null;
+    }
+
+    // whether the input is empty
+    atEnd() {
+        return this._lines.length === 0;
+    }
+
+    // returns the unparsed (yet) string
+    remaining() {
+        return this._lines.join('\n');
+    }
+}
+
+// Converts the raw PR message into the staged commit message.
+class CommitMessage
+{
+    constructor(rawPr) {
+        // the (required) commit message title
+        this._title = rawPr.title + ' (#' + rawPr.number + ')';
+        // the (optional) message body without optional trailer, may be missing
+        this._body = undefined;
+        // The (optional) finalizing message part with GitHub-related attributes,
+        // separated from the body by an empty line.
+        this._trailer = undefined;
+        // the (optional) Authored-by meta information extracted from the PR description header
+        this._author = undefined;
+        this._error = undefined; // Error object in a case of a parsing problem
+        // message attributes supported by the bot
+        this._attributes = ["Authored-by", "Co-authored-by"];
+
+        if (rawPr.body !== undefined && rawPr.body !== null) {
+            this._body = rawPr.body.replace(/\r+\n/g, '\n').trimEnd();
+            this._parse();
```

Comment:
I would not store raw/unparsed _body because it creates the same set of problems we are trying to solve by adding this class: The answer to the "What is a _body?" question becomes very complex.

```suggestion
            const trimmedPrDescripton = rawPr.body.replace(/\r+\n/g, '\n').trimEnd();
            this._parse(trimmedPrDescripton);
```